lib/libthr/thread/thr_create.c

```diff
@@ -156,10 +156,11 @@ _pthread_create(pthread_t * __restrict thread,
 		 * c18n: Always block all signals when creating a new thread to
 		 * allow RTLD to set up the environment to handle signals.
 		 */
-		create_suspended = 1;
-#else
-		create_suspended = 0;
+		if (_rtld_c18n_is_enabled())
+			create_suspended = 1;
+		else
 #endif
+			create_suspended = 0;
```

Comment:
Somewhat an upstream problem, but I feel like the create_suspended code would be cleaner if it was a bool initialed to false on declaration and set to true in these two cases.